### ISSUE TITLE
Warnings when rebuilding search index

### ISSUE THREAD

**Issue by ctgraham** [pkp/pkp-lib#4720]
**Describe the bug**
https://forum.pkp.sfu.ca/t/using-rebuildsearchindex-php-in-ojs-3-1-2/53044

> This week I’m finishing my upgrade from OJS 2.4.7 to 3.1.2, and still have some doubts about rebuildSearchIndex.php. Runing it in this version gets me thousands of:
```
PHP Warning:  Illegal offset type in /home/ojs/rc-3.1.2/lib/pkp/classes/search/SubmissionSearchDAO.inc.php on line 48
PHP Warning:  Illegal offset type in isset or empty in /home/ojs/rc-3.1.2/lib/pkp/classes/search/SubmissionSearchDAO.inc.php on line 26
PHP Warning:  Illegal offset type in /home/ojs/rc-3.1.2/lib/pkp/classes/search/SubmissionSearchDAO.inc.php on line 48
PHP Warning:  Illegal offset type in isset or empty in /home/ojs/rc-3.1.2/lib/pkp/classes/search/SubmissionSearchDAO.inc.php on line 26
```

**To Reproduce**
1. In a journal with one or more articles, run `tools/rebuildSearchIndex.php`
2. PHP warnings will occur for each locallized setting in each article

**What application are you using?**
 OJS 3.1.2.  Should also affect OMP 3.1.2

**Additional information**
The actual of the warnings appears to be the fact that `PKPString` now returns `Stringy\Stringy` objects, and then these are passed unchecked as array indexes.

https://github.com/pkp/pkp-lib/blob/41216c3b54791beb0199eae2bb4c849487ecd45d/classes/search/SubmissionSearchIndex.inc.php#L54
calls
https://github.com/pkp/pkp-lib/blob/41216c3b54791beb0199eae2bb4c849487ecd45d/classes/core/PKPString.inc.php#L160-L168
which is then routed through
https://github.com/pkp/ojs/blob/9a9b7978a0e8f2a03d444d735b533d896d75c286/classes/search/ArticleSearchIndex.inc.php#L304
to
https://github.com/pkp/pkp-lib/blob/41216c3b54791beb0199eae2bb4c849487ecd45d/classes/search/SubmissionSearchDAO.inc.php#L26
and
https://github.com/pkp/pkp-lib/blob/41216c3b54791beb0199eae2bb4c849487ecd45d/classes/search/SubmissionSearchDAO.inc.php#L48

**Comment by asmecher** [comment]
@ctgraham, did https://github.com/pkp/pkp-lib/issues/4722 fix this?

**Comment by ctgraham** [comment]
#4722 resolves the warnings, but not the inconsistent use of localized return values I found when investigating the warnings.  I'll adjust the title.

**Comment by ctgraham** [comment]
Branching the documentation issue to a new issue.

This issue is resolved by #4722.
